### ISSUE TITLE
2021.04.02 - Importer Issues - Scaling and Rendering

### ISSUE THREAD

**Issue by michaeloldman** [TheGameCreators/GameGuruRepo#1059]
Incorrect Scaling
The importer does not import models at correct scale
- Tried three models 
- when first loads model it is the correct scale
- save and load the model - incorrect scale
- 
![Importer_Woes_01](https://user-images.githubusercontent.com/12246318/113462077-f0a93f80-946a-11eb-8862-8f6cfea1fbcc.png)

Incorrect Rendering
(this was also in previous builds)
- Importer does not render all parts of model correctly
- - maybe post render error? 
- - (rotate this model and same parts that render correctly in one direction, fail when rotated to this position)

![Importer_Woes_02](https://user-images.githubusercontent.com/12246318/113462134-3108bd80-946b-11eb-9914-a4994d227450.png)

System Specs:
Operating System: Windows 10 Pro 64-bit (10.0, Build 19042) (19041.vb_release.191206-1406)
Language: English (Regional Setting: English)
System Model: Z390 UD
Processor: Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 CPUs), ~3.6GHz
Memory: 32768MB RAM
Windows Dir: C:\WINDOWS
DirectX Version: DirectX 12
Card name: NVIDIA GeForce GTX 1060 6GB
Dedicated Memory: 6052 MB
Current Mode: 1920 x 1080 (32 bit) (60Hz)
Card name: NVIDIA GeForce GTX 1060 6GB
Dedicated Memory: 6052 MB
Current Mode: 1920 x 1080 (32 bit) (60Hz)
Card name: NVIDIA GeForce GTX 1060 6GB
Dedicated Memory: 6052 MB
Current Mode: 1920 x 1080 (32 bit) (60Hz)

**Comment by michaeloldman** [comment]
Issue remains in Build 2021.04.16

**Comment by michaeloldman** [comment]
Build 2021.05.07
Scaling is no longer an issue.
Render issues remain.

**Comment by LeeBamberTGC** [comment]
This one is a stinker.  Could very well be the difference between loading an X file and importing via AssImp, and all the many steps between the file and the engine which could twizzle the normal behavior.  I have the Reactor Room 01 entity and had a good look but cannot see the seam where the normal is flipped:

![image](https://user-images.githubusercontent.com/35840583/117872535-b10a1a80-b296-11eb-95d8-291239864742.png)


Can you maybe tell me the name of the entity that shows it, or even better resend the model on its own with a shot when you have time, thanks!

**Comment by rvanner** [comment]
@michaeloldman Is this still a problem for you? Can you reply to Lee's question?

**Comment by michaeloldman** [comment]
@rvanner Sorry missed that.
Normal issues now seam to show up only where there are lights. 
Here is a pic I just snapped in current build.
![Normals Issue 2021 08 20](https://user-images.githubusercontent.com/12246318/130353285-4a785cad-540c-4be1-8667-31f4b2cdf113.png)

This can be seen in multiple paces in the demo I sent your awhile back.

Please see this issue #984

**Comment by rvanner** [comment]
@michaeloldman Can you send a cut down example of this problem? Ideally I would like you to send me a level with two objects, one showing correctly and one that is wrong. That would help us to quickly identify the issue. Lee's initial view is that you have the normals wrong and are expecting the engine to convert them. While that would be okay in Classic it's not in Max (as Wicked must have them in the exact layout).

**Comment by michaeloldman** [comment]
@rvanner Left Message on Discord :)
Some things to keep in mind
- it seams to be MAX no longer likes mirrors - which is where the errors are showing up (on the mirrored side)
- the models show correctly in Substance Painter
- MAX used to show them correctly before repo and assimp integration
- tis not end of world if can't be fixed - I have the bin next to me :)

**Comment by rvanner** [comment]
@michaeloldman With so many other things taking the team's attention I think we will have to leave this for later. If there's a way around it for now then please take that route. I do want MAX to deal with many situations and I think we will be tweaking the importer for months to come to make it the best it can be. I will demote this to medium for now. I hope you see the sense in our prioritising of jobs.

**Comment by michaeloldman** [comment]
@rvanner no problems, I understand. :)

I was waiting to see if it was fixable before I finally decided to bin the current Racha models. Not using mirrors is a waste of resources to me. Just about every model I have checked out on SkechFab uses them, I have also looked at similar issues reported on Unity forums, but none of those fixes worked either, so it seams the only way round this issue is not to use mirrors in MAX.

However, now that we can easily use multi-textured models I will seriously contemplate redoing them without the use of mirrors.

Ton of work, but it is what it is. On the other hand I may just not use the Racha Level at all yet.

Anyhow no worries, the team has much to do already.
